src/main/host/thread_ptrace.c

```diff
@@ -1079,41 +1079,63 @@ bool threadptrace_isRunning(Thread* base) {
     return false;
 }
 
-void threadptrace_terminate(Thread* base) {
-    ThreadPtrace* thread = _threadToThreadPtrace(base);
-
-    /* make sure we cleanup circular refs */
-    if (thread->sys) {
-        syscallhandler_unref(thread->sys);
-        thread->sys = NULL;
-    }
-
-    if (!threadptrace_isRunning(base)) {
+static void _threadptrace_terminateHelper(ThreadPtrace* thread) {
+    // Nothing to do if we already exited
+    if (!threadptrace_isRunning(&thread->base)) {
         return;
     }
 
-    // need to kill() and not ptrace() since the process may not be stopped
-    if (kill(thread->base.nativeTid, SIGKILL) < 0) {
-        warning("kill %d: %s", thread->base.nativePid, g_strerror(errno));
+    // First let's make sure that the thread is detached
+    if (!thread->needAttachment) {
+        // Still attached, detach before killing
```

Comment:
Why not call `_threadptrace_doDetach`? (If it's just because of the utility_assert in there about the childState, we could just remove it)